src/node_file.cc

```diff
@@ -1645,49 +1546,28 @@ static void Unlink(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
 
   const int argc = args.Length();
-  CHECK_GE(argc, 2);
+  CHECK_GE(argc, 1);
 
   BufferValue path(env->isolate(), args[0]);
   CHECK_NOT_NULL(*path);
   THROW_IF_INSUFFICIENT_PERMISSIONS(
       env, permission::PermissionScope::kFileSystemWrite, path.ToStringView());
 
-  FSReqBase* req_wrap_async = GetReqWrap(args, 1);
-  if (req_wrap_async != nullptr) {
+  if (argc == 2) {  // unlink(path, req)
```

Comment:
I think this should be the other way around. If you pass 3 arguments to this function, (path, req, null), I expect to go through the AsyncCall path, not to the sync path one

Reply:
Done.